test/Dapr.Client.Test/InvokeApiTest.cs

```diff
@@ -413,6 +423,271 @@ public async Task InvokeMethodAsync_WithReturnTypeAndData_WithNonDefaultVerb_Wit
+                client.Call<InvokeResponse>()
+                .SetResponse(invokeResponse)
+                .AddHeader("some-header", "some-value")
+                .Build();
+
+
+            var trailers = new Metadata();
+             const string grpcErrorInfoReason = "Insufficient permissions";
+            const int grpcErrorInfoDetailHttpCode = 500;
+            const string grpcErrorInfoDetailHttpErrorMsg = "no permissions";
+            int grpcStatusCode = Convert.ToInt32(StatusCode.PermissionDenied);
+            const string grpcStatusMessage = "Bad permissions";
+            var details = new Google.Rpc.Status
+            {
+                Code = grpcStatusCode,
+                Message = grpcStatusMessage,
+            };
+
+            var errorInfo = new Google.Rpc.ErrorInfo
+            {
+                Reason = grpcErrorInfoReason,
+                Domain = "dapr.io",
+            };
+            errorInfo.Metadata.Add("http.code", grpcErrorInfoDetailHttpCode.ToString());
+            errorInfo.Metadata.Add("http.error_message", grpcErrorInfoDetailHttpErrorMsg);
+            details.Details.Add(Google.Protobuf.WellKnownTypes.Any.Pack(errorInfo));
+
+
+            var entry = new Metadata.Entry("grpc-status-details-bin", Google.Protobuf.MessageExtensions.ToByteArray(details));
+            trailers.Add(entry);
+
+            const string rpcExceptionMessage = "No access to app";
+            const StatusCode rpcStatusCode = StatusCode.PermissionDenied;
+            const string rpcStatusDetail = "Insufficient permissions";
+
+            var rpcException = new RpcException(new Status(rpcStatusCode, rpcStatusDetail), trailers, rpcExceptionMessage);
+
+
+            // Setup the mock client to throw an Rpc Exception with the expected details info
+            client.Mock
+                .Setup(m => m.InvokeServiceAsync(It.IsAny<Autogen.Grpc.v1.InvokeServiceRequest>(), It.IsAny<CallOptions>()))
+                .Throws(rpcException);
+
+            try
+            {
+                var body = new Request() { RequestParameter = "Hello " };
+                await client.DaprClient.InvokeMethodWithResponseHeadersAsync<Request, Response>("test", "testMethod", body);
+                Assert.False(true);
+            }
+            catch(ServiceInvocationException<Request, Response> ex)
+            {
+                ex.Message.Should().Be("Exception while invoking testMethod on appId:test");
+                ex.InnerException.Message.Should().Be(rpcExceptionMessage);
+
+                ex.Response.GrpcStatusInfo.GrpcErrorMessage.Should().Be(grpcStatusMessage);
+                ex.Response.GrpcStatusInfo.GrpcStatusCode.Should().Be(grpcStatusCode);
+                ex.Response.GrpcStatusInfo.InnerHttpErrorMessage.Should().Be(grpcErrorInfoDetailHttpErrorMsg);
+                ex.Response.GrpcStatusInfo.InnerHttpStatusCode.Should().Be(grpcErrorInfoDetailHttpCode);
+                ex.Response.HttpStatusCode.Should().BeNull();
```

Comment:
I'm a little confused about this given that `http.code` is set in the code above - what is this intended to test exactly?

Reply:
When Dapr sidecar receives HTTP error code = 500 and error message in the response body, it maps the http status code to grpc status code. It preserves the original error code (as metadata key = http.code) and error message (as metadata key = http.error_message) and sends it in the trailer. This test is simulating the response from Dapr sidecar and checking whether the dotnet sdk is able to parse it correctly. This is similar to the sample code provided to Zeiss in https://github.com/dapr/dotnet-sdk/issues/399

Reply:
so I think I'm still confused. If this test **represents** an HTTP 500 returned by an HTTP callee - then why is `response.HttpStatusCode` null?

Reply:
I have updated the tests to separate the ones for GRPC callee and HTTP callee. So this should be null for grpc callee. This comment is referencing the old code?